### PR TITLE
Feat(medusa): Integration testing of cli

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- add github action to test cli methods: new | migrations run | seed | user | start | develop
  - action runs on pull request

**How**
- With an action that for both a sqlite and a postgres backend does the following: 
  - sets up a medusa server `medusa new`
  - runs `medusa-dev --force-install`
  - runs migrations `medusa migrations run`
  - seeds the database `medusa seed`
  - creates a user with `medusa user`
  - test that the user can log in and that products are seeded for `medusa develop` and `medusa start`

### CODE REVIEW

**Review by srindom — APPROVED** [review]
LGTM! Really nice addition - one suggestion could be to also add a test for the npx command, but that can be a separate PR.

File: .github/actions/test-server/action.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+runs:
+  using: "composite"
+  steps:
+    - name: Wait for live server response
+      shell: "bash"
+      run: ./integration-tests/scripts/cli/wait-for-server-live.sh
+      # run: ./node_modules/@medusajs/medusa-cli/scripts/__tests__/wait-for-server-live.sh
+
+      #'curl -X POST http://localhost:9000/admin/auth -H "Content-Type: application/json" -d ''{"email":"test@test.com", "password":"password"}'''
```
Comment (olivermrbl) [inline .github/actions/test-server/action.yml:26]: todo: Clean up
```suggestion
```

File: .github/actions/test-server/action.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    - name: Log in with user
+      shell: "bash"
+      run: ./integration-tests/scripts/cli/login.sh ${{ inputs.email }} ${{ inputs.password }}
+      # run: ./node_modules/@medusajs/medusa-cli/scripts/__tests__/login.sh ${{ inputs.email }} ${{ inputs.password }}
+
+      # "curl -X GET http://localhost:9000/store/products"
```
Comment (olivermrbl) [inline .github/actions/test-server/action.yml:32]: todo: Clean up
```suggestion
```

File: .github/actions/test-server/action.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    - name: GetProducts
+      shell: "bash"
+      run: ./integration-tests/scripts/cli/get-products.sh ${{ inputs.pathToSeedData }}
+      # run: ./node_modules/@medusajs/medusa-cli/scripts/__tests__/get-products.sh ${{ inputs.pathToSeedData }}
```
Comment (olivermrbl) [inline .github/actions/test-server/action.yml:36]: todo: Clean up
```suggestion
```

**Review by olivermrbl — APPROVED** [review]
LGTM, this is super nice!

**Pull request merged**